### PR TITLE
Upload collection

### PR DESCRIPTION
**Pull request by MilanPospisil**
https://issues.redhat.com/browse/AAH-2455

Collections - Upload collection button.

PR added selection of repository. Previously, upload was done into staging repository. There is radio button that allows switching between staging repos and repos without any pipeline (so nor staging, approved or rejected).

Then user can select repository in table and file. Selected collection file will be uploaded to selected repo.

The namespace parameter can be added to query string. This can be used in link in namespace detail to upload collection to namespace. The namespace parameter will be then matched with namespace of uploaded collection and if it is not equal, the uploading will fail.

Before:
image

After:
image

### CODE REVIEW

File: frontend/hub/collections/UploadCollection.tsx [hunk 1]
```diff
@@ -38,6 +61,75 @@ export function UploadCollectionByFile() {
   const repositories = useRepositories();
   const navigate = useNavigate();
   const onCancel = () => navigate(-1);
+  const toolbarFilters = useRepoFilters();
+  const tableColumns = useRepositoriesColumns();
+  const [onlyStaging, setOnlyStaging] = useState(true);
+  const [selectedRepo, setSelectedRepo] = useState<{ name: string } | null>(null);
```
Comment (nixocio) [inline frontend/hub/collections/UploadCollection.tsx:67]: Are we interested in the name only or the whole repository data?

File: frontend/hub/collections/UploadCollection.tsx [hunk 2]
```diff
@@ -38,6 +61,75 @@ export function UploadCollectionByFile() {
...
+      ? {
+          pulp_label_select: 'pipeline=staging',
+        }
+      : {
+          pulp_label_select: '!pipeline',
+        },
+  });
+
+  useEffect(() => {
+    setSelectedRepo({ name: 'staging' });
```
Comment (nixocio) [inline frontend/hub/collections/UploadCollection.tsx:84]: Here we set the name only, the below we set the whole repository data.

File: frontend/hub/collections/UploadCollection.tsx [hunk 3]
```diff
@@ -38,6 +61,75 @@ export function UploadCollectionByFile() {
...
+        ></Radio>
+        <div>
+          {!onlyStaging && (
+            <>{t`Please note that those repositories are not filtered by permission, if operation fail, you don't have it.`}</>
+          )}
+        </div>
+
+        <PageTable<Repository>
+          onSelect={(repo) => {
+            setSelectedRepo(repo);
```
Comment (nixocio) [inline frontend/hub/collections/UploadCollection.tsx:116]: Adding the whole repository data.

File: frontend/hub/collections/UploadCollection.tsx [hunk 4]
```diff
@@ -84,3 +180,46 @@ export function UploadCollectionByFile() {
...
+
+export function useRepositoriesColumns() {
+  const { t } = useTranslation();
+  const tableColumns = useMemo<ITableColumn<Repository>[]>(
+    () => [
+      {
+        header: t('Name'),
+        sort: 'name',
+        cell: (repository) => (
+          <>
```
Comment (nixocio) [inline frontend/hub/collections/UploadCollection.tsx:192]: We can remove this fragment.

File: frontend/hub/collections/UploadCollection.tsx (hunk 1 above)
Comment (MilanPospisil) [reply frontend/hub/collections/UploadCollection.tsx:67]: For now, only name. But i think its better to pass { name : string }, so we will be able to easily add another data if needed in future.

Comment (MilanPospisil) [reply frontend/hub/collections/UploadCollection.tsx:67]: Future is now, I need to add pulp_href :D

**Review by himdel — APPROVED** [review]

File: frontend/hub/collections/UploadCollection.tsx [hunk 5]
```diff
@@ -38,6 +67,98 @@ export function UploadCollectionByFile() {
   const repositories = useRepositories();
   const navigate = useNavigate();
   const onCancel = () => navigate(-1);
+  const toolbarFilters = useRepoFilters();
+  const tableColumns = useRepositoriesColumns();
+  const [onlyStaging, setOnlyStaging] = useState(true);
+  const [selectedRepo, setSelectedRepo] = useState<{ name: string; pulp_href: string } | null>(
+    null
+  );
+  const distroGetRequest = useGetRequest<{ results: Distribution[] }>();
```
Comment (MilanPospisil) [inline frontend/hub/collections/UploadCollection.tsx:76]: PulpItemsResponse, HubItemsRespons deals with returned type.

**Pull request merged**